common/src/hash.rs

```diff
@@ -136,12 +137,12 @@ pub fn hash_float(value: f64) -> PyHash {
     };
     x = ((x << e) & MODULUS) | x >> (BITS32 - e);
 
-    fix_sentinel(x as PyHash * value.signum() as PyHash)
+    Some(fix_sentinel(x as PyHash * value.signum() as PyHash))
 }
 
 pub fn hash_complex(value: &Complex64) -> PyHash {
-    let re_hash = hash_float(value.re);
-    let im_hash = hash_float(value.im);
+    let re_hash = hash_float(value.re).unwrap();
+    let im_hash = hash_float(value.im).unwrap();
```

Comment:
This will cause panic if `complex(float('nan'), float('nan'))` is passed to builtin hash function.
```python
>>>>> hash(complex(float('nan'), float('nan')))
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', common/src/hash.rs:144:40
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
`re_hash` and `im_hash` also need to have `hash_pointer` value if the given value is `nan`.